### PR TITLE
`boundary_partitionedges`

### PR DESCRIPTION
**Pull request by JoeyT1994**
New function `boundary_partitionedges(pg::PartitionedGraph, partitionvertices::Vector{<:PartitionVertex})` which returns all the `partitionedges` (in `PartitionEdge` format) incoming to the region specified by `partitionvertices`. Tests are also added.

### CODE REVIEW

File: src/Graphs/partitionedgraphs/partitionedgraph.jl [hunk 1]
```diff
@@ -94,12 +94,26 @@ function edges(pg::PartitionedGraph, partitionedges::Vector{<:PartitionEdge})
...
+)
+  return PartitionEdge.(
+    boundary_edges(partitioned_graph(pg), parent.(partitionvertices); kwargs...)
+  )
+end
+
+function boundary_partitionedges(
+  pg::PartitionedGraph, partitionvertex::PartitionVertex; kwargs...
+)
+  return boundary_partitionedges(pg, PartitionVertex[partitionvertex]; kwargs...)
```
Comment (mtfishman) [inline src/Graphs/partitionedgraphs/partitionedgraph.jl:108]:
```suggestion
  return boundary_partitionedges(pg, [partitionvertex]; kwargs...)
```
Julia should already infer the type is `Vector{<:PartitionVertex}`.

File: src/NamedGraphs.jl [hunk 2]
```diff
@@ -187,6 +187,7 @@ export NamedGraph,
   partitionvertex,
   partitionvertices,
   partitioned_vertices,
+  boundary_partitionedges,
```
Comment (mtfishman) [inline src/NamedGraphs.jl:190]: Let's avoid exporting for now and reassess exports going forward.

For example, ideally we would have a `PartitionedGraphs` module, which then exports partition-related functions. So better off not exporting so we don't have to backtrack later when we move to that design.

**Comment by mtfishman** [comment]
Could you add a test for `copy(::PartitionedGraph)`?

**Comment by JoeyT1994** [comment]
Sure I just added that.

**Comment by mtfishman** [comment]
Thanks!

**Comment by JoeyT1994** [comment]
Can we move to a new version of NamedGraphs following this PR? Just so the tests for the ITensorNetworks PR: https://github.com/mtfishman/ITensorNetworks.jl/pull/139 pass

**Comment by mtfishman** [comment]
Thanks for the reminder, registering it now.

**Pull request merged**